### PR TITLE
Markdown parser using markdown-it-py

### PR DESCRIPTION
**Pull request by miteshashar**

### CODE REVIEW

File: funnel/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"""Utility functions."""
+# flake8: noqa
+
+from .markdown import markdown
```
Comment (jace) [inline funnel/utils/__init__.py:4]: Always do `import *` in `__init__.py`, and declare an `__all__` list in the respective modules if you don't want to export everything.

File: funnel/utils/markdown/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+"""Markdown parser using markdown-it-py."""
+# flake8: noqa
+
+from .base import markdown
```
Comment (jace) [inline funnel/utils/markdown/__init__.py:4]: Same here, `import *`

File: funnel/utils/markdown/base.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+from typing_extensions import Protocol
+
+from coaster.utils import make_name
+from coaster.utils.text import VALID_TAGS, normalize_spaces_multiline, sanitize_html
+
+MARKDOWN_HTML_TAGS = deepcopy(VALID_TAGS)
+
+
+# --- Typing Protocol for markdown-it-py plugins ---------------------------------------
+class MarkdownItPluginProtocol(Protocol):  # pylint: disable=too-few-public-methods
```
Comment (jace) [inline funnel/utils/markdown/base.py:18]: This disable should not be necessary as it's already in the global list in `pyproject.toml`.

File: funnel/utils/markdown/base.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+#       Port from https://github.com/markdown-it/markdown-it-sup
+#       Port from https://github.com/markdown-it/markdown-it-ins
+# TODO: emoji, (mark => highlight, inlinehilite)
+#       Port from https://github.com/markdown-it/markdown-it-emoji
+#       Port from https://github.com/markdown-it/markdown-it-mark
+#       Evaluate:
+#       https://www.npmjs.com/search?q=highlight%20keywords%3Amarkdown-it-plugin
+
+
+default_markdown_extensions_html: Mapping[str, Type[MarkdownItPluginProtocol]] = {
```
Comment (jace) [inline funnel/utils/markdown/base.py:39]: The `Type[]` wrapper should not be necessary. You can use `Dict[str, MarkdownItPluginProtocol]` as this is an explicit dict and not an abstract Mapping.

**Review by jace — CHANGES_REQUESTED** [review]

**Review by miteshashar — COMMENTED** [review]
Need resolution on this to continue merging.

**Pull request merged**